### DISCUSSION TITLE
Automatic instrumentation vs manual - when and how?

### DISCUSSION THREAD

**Discussion by dominikjeske** [open-telemetry/opentelemetry-dotnet#5022] [General]
Open telemetry is changing rapidly, and I'm confused about how to configure instrumentation of my apps. For a while we have option of Automatic instrumentation but I can't find anywhere when this scenario is better then manual. For example how it affects performance? Another question is how exactly is this mechanism working? I see docs about configuration but I can't find any on how automatic instrumentation actually works. How it magically reads metrics from process, what techniques it uses, is it working out of process etc. Is it applicable to all scenarios? For example documentations about IIS https://opentelemetry.io/docs/instrumentation/net/automatic/ says about .NET Framework but nothing about .NET Core?

**Comment by vishweshbankwar** [comment]
@dominikjeske https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation would be the right place to ask questions about auto-instrumentation.

**Reply by dominikjeske** [reply]
@vishweshbankwar ok I posted question there -> https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/discussions/3068
